mw-webapp/src/pages/authPage/AuthPage.tsx

```diff
@@ -1,14 +1,36 @@
-import {Link} from "react-router-dom";
-import {useAuth} from "src/utils/useAuth";
+import {User} from "firebase/auth";
+import {useEffect, useState} from "react";
+import {Link, useNavigate} from "react-router-dom";
+import {Button} from "src/component/button/Button";
+import {getNewUserCredentials} from "src/utils/auth/getNewUserCredentials";
+import {handleUserAuthState} from "src/utils/auth/handleAuthState";
+import {handleLogIn} from "src/utils/auth/handleLogIn";
+import {handleLogOut} from "src/utils/auth/handleLogOut";
 
 export const AuthPage = () => {
-  const handleGoogleSignIn = useAuth();
+  const [user, setUser] = useState<User | null>(null);
+  const navigate = useNavigate();
+
+  useEffect(() => {
+    return () => {
+      handleUserAuthState(setUser);
```

Comment:
this callback wil be invoked after unmounting  Auth page. Why do you need to handle auth state in this case and what does it mean. Use more approproate abstraction or logic.